task10/script.js

```diff
@@ -0,0 +1,516 @@
+
+
+    getPage(skip = 0, top = 10, filterConfig) {
+        let arr = [];
+        let toDate;
+        let fromDate;
+        let author;
+        if (typeof (filterConfig) !== 'undefined') {
+            toDate = new Date(filterConfig.to);
+            fromDate = new Date(filterConfig.from);
+            if (isNaN(toDate)) {
+                toDate = this._posts[this._posts.length - 1].createdAt;
+            }
+
+            if (isNaN(fromDate)) {
+                fromDate = this._posts[0].createdAt;
+            }
+            author = filterConfig.author;
+        } else {
+            toDate = this._posts[this._posts.length - 1].createdAt;
+            fromDate = this._posts[0].createdAt;
+        }
+        let skipCount = 0;
+        let topCount = 0;
+
+        if (typeof (author) === 'string' && author !== '') {
+            for (let i = this._posts.length - 1; i >= 0; i--) {
+                if (fromDate <= this._posts[i].createdAt && toDate >= this._posts[i].createdAt && author === this._posts[i].author) {
+                    if (skipCount >= skip && topCount < top) {
+                        arr.push(this._posts[i]);
+                        topCount++;
+                    }
+                    skipCount++;
+                }
+                if (topCount >= top) {
+                    break;
+                }
+            }
+        } else if (typeof (author) !== 'string' || author === '') {
+            for (let i = this._posts.length - 1; i >= 0; i--) {
+                if (fromDate <= this._posts[i].createdAt && toDate >= this._posts[i].createdAt) {
+                    if (skipCount >= skip && topCount < top) {
+                        arr.push(this._posts[i]);
+                        topCount++;
+                    }
+                    skipCount++;
+                }
+                if (topCount >= top) {
+                    break;
+                }
+            }
+        }
+
+        return arr;
+
+    }
+
+    addAll(posts) {
+        let arr = [];
+        for (let i = 0; i < posts.length; i++) {
+            posts[i].createdAt = new Date(posts[i].createdAt);
+            if (this._validate(posts[i])) {
+                this._posts.push(posts[i]);
+            } else {
+                arr.push(posts[i]);
+            }
+        }
+        this.save();
+        return arr;
+    }
+
+    save() {
+        localStorage.setItem("posts", JSON.stringify(this._posts));
+    }
+
+}
+
+let postCollection = new PostCollection();
+postCollection.addAll(posts);
+
+class View {
+
+    static build(post) {
+        let postDiv = document.createElement("div");
+        postDiv.className = "postDiv";
+        postDiv.id = post.id;
+
+        let infoDiv = document.createElement("div");
+
+        if (user.name === post.author) {
+
+            let deleteA = document.createElement("a");
+            deleteA.className = "editDelete";
+            deleteA.setAttribute("onclick", "deletePost(this.parentNode.parentNode.id)");
```

Comment:
1) Don't use setAttribute for it, it could be `deleteA.onclick = () => deletePost(this.parentNode.parentNode.id)`
2) > Не обрабатывайте события динамического контента (твитов) непосредственно на динамических элементах, используйте [делегирование событий](https://learn.javascript.ru/event-delegation).